### PR TITLE
Add FSGroup check to e2e of applicable volumes

### PR DESCRIPTION
**Pull request by ghost**
This adds a check for FSGroup to the volumes which support ownership management.

@pmorie @jsafrane

### CODE REVIEW

File: test/e2e/volumes.go [hunk 1]
```diff
@@ -167,16 +168,19 @@ func volumeTestCleanup(client *client.Client, config VolumeTestConfig) {
 		glog.Warningf("Failed to delete client pod: %v", err)
 		expectNoError(err, "Failed to delete client pod: %v", err)
 	}
-	err = podClient.Delete(config.prefix+"-server", nil)
-	if err != nil {
-		glog.Warningf("Failed to delete server pod: %v", err)
-		expectNoError(err, "Failed to delete server pod: %v", err)
+
+	if config.serverImage != "" {
```
Comment (jsafrane) [inline test/e2e/volumes.go]: Why was this check added? I don't see any change, that would lead to `config.serverImage == ""`

**Comment by jsafrane** [comment]
One minor nit, otherwise LGTM.

File: test/e2e/volumes.go (hunk 1 above)
Comment (swagiaal) [inline test/e2e/volumes.go]: > Why was this check added?

The cinder test does not create a server pod. It then later fails while trying to delete the server pod. This check fixes that

**Comment by jsafrane** [comment]
With the last @swagiaal comment, LGTM without any nits.

**Comment by swagiaal** [comment]
@k8s-bot unit test this please.

**Comment by swagiaal** [comment]
@ixdy could you take a look at this ?

**Comment by ixdy** [comment]
@swagiaal sorry for late reply - things have been a little busy here tracking flaky tests.

is there anyone more familiar with FSGroup stuff who might be a better reviewer? @pmorie @thockin @kubernetes/goog-cluster

**Comment by ixdy** [comment]
@k8s-bot test this please

File: test/e2e/volumes.go [hunk 2]
```diff
@@ -285,6 +304,11 @@ func injectHtml(client *client.Client, config VolumeTestConfig, volume api.Volum
 					},
 				},
 			},
+			SecurityContext: &api.PodSecurityContext{
```
Comment (pmorie) [inline test/e2e/volumes.go]: I'm wondering if this test should be tagged with SELinux.  It's debatable; I don't think I would normally want to run this with 'stuff that exercises core SELinux functionality in kube', but I definitely want to test it when I run absolutely everything that touches SELinux.

File: test/e2e/volumes.go (hunk 1 above)
Comment (pmorie) [inline test/e2e/volumes.go]: Add a comment explaining the cinder situation

**Comment by pmorie** [comment]
@swagiaal one nit re: comment, otherwise LGTM

**Pull request merged**